docs/reference/src/literal-expressions.adoc

```diff
@@ -34,25 +34,17 @@ LITERAL_SUFFIX : XID_Start XID_Continue*
 
 A numeric literal can be written using three bases:
 
-1. Decimal literal starts with a decimal digit and then a mixture of decimal digits and underscores.
-2. Hexadecimal literal starts with the character sequence `0x` and continues as a mixture of hex
-   digits and underscores.
-3. Binary literal starts with the character sequence `0b` and continues as a mixture of binary
-   digits and underscores.
+1. Decimal literal starts with a decimal digit and then decimal digits.
+2. Hexadecimal literal starts with the character sequence `0x` and continues with hex digits.
+3. Binary literal starts with the character sequence `0b` and continues with binary digits.
 
-Underscore character (`_`) is only a visual separator, and it has no influence on the number's
-value.
-
-> **Note**: Character sequence `_1234` is a valid identifier, not an numeric literal.
+A numeric literal may be followed (immediately, without any spaces) by an underscore character (`_`)
+and then a _literal suffix_, which forcibly sets the type of the literal.
+The literal suffix is any valid identifier, but only selected values are semantically correct.
 
 > **Note**: Floating-point numbers are not supported.
 
-A numeric literal may be followed (immediately, without any spaces) by a _literal suffix_, which
-forcibly sets the type of the literal.
-The literal suffix is any valid identifier which does not start with an underscore (`_`), but only
-selected values are semantically correct.
-
-The type of _unsuffixed number literal_ is determined by type inference:
+The type of an _unsuffixed number literal_ is determined by type inference:
```

Comment:
Is it? currently it's just felt. I think it is best to couple the doc with the curent version, and when we change it, change the doc as well.